### DISCUSSION TITLE
Setting Authorization Headers

### DISCUSSION THREAD

**Discussion by LongBeachHXC** [open-telemetry/opentelemetry-java#6405] [Q&A] [answered]
I'm getting the following error message when attempting to configure an authorization header via an environment variable. The env var I'm using is 'OTEL_EXPORTER_OTLP_HEADERS'. I'm assuming the error message is because `Api-Token` isn't an official http authorization scheme as listed on [mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes). The format I'm using is below and I've tried several different variations. All report the same error.

Some clarification on this would be greatly appreciated.

```
# Header format
Authorization="Api-Token <token>"
Authorization=Api-Token <token
Authorization="Api-Token%20<token>"
Authorization=Api-Token%20<token>
```
```
#Error Message

io.opentelemetry.exporter.internal.http.HttpExporter - Failed to export spans. Server responded with HTTP status code 401. Error message: Authorization header contains unsupported authorization scheme '"Api-Token'
```

**Comment by jack-berg** [comment]
`OTEL_EXPORTER_OTLP_HEADERS` values should be percent encoded, as described in the [spec](OTEL_EXPORTER_OTLP_HEADERS). But the implementation is lenient and spaces which aren't percent encoded don't cause problems. 

Here's a test showing what the exporter is configured with for the different `OTEL_EXPORTER_OTLP_HEADERS` you included:
```
  @ParameterizedTest
  @MethodSource("source")
  void headerParsing(String envVar, Map<String, String> headers) {
    try (SpanExporter spanExporter =
        provider.createExporter(
            DefaultConfigProperties.createFromMap(ImmutableMap.of(
                "otel.exporter.otlp.headers", envVar,
                "otel.exporter.otlp.protocol", "http/protobuf")))) {
      headers.forEach(
          (key, value) -> {
            verify(httpBuilder).addHeader(key, value);
          });
    }
  }

  static Stream<Arguments> source() {
    return Stream.of(
        Arguments.of("Authorization=\"Api-Token token\"", ImmutableMap.of("Authorization", "\"Api-Token token\"")),
        Arguments.of("Authorization=Api-Token token", ImmutableMap.of("Authorization", "Api-Token token")),
        Arguments.of("Authorization=\"Api-Token%20token\"", ImmutableMap.of("Authorization", "\"Api-Token token\"")),
        Arguments.of("Authorization=Api-Token%20token", ImmutableMap.of("Authorization", "Api-Token token"))
    );
  }
```
The error message seems to indicate you're using a value which includes double quotes. This isn't necessary.
> Authorization header contains unsupported authorization scheme '"Api-Token'

**Reply by LongBeachHXC** [reply, marked as answer]
I really appreciate your help. Two things helped here:

1. Percent encoding my value
2. Removing the double quotes

Just an aside, the service I am authing against doesn't like the `%20` encoding. I changed the encoding to use the `+` sign and everything worked as expected.

Thank you @jack-berg
